src/app/components/AccountMenu/index.tsx

```diff
@@ -83,9 +83,12 @@ function AccountMenu({ title, subtitle, showOptions = true }: Props) {
                   selectAccount(accountId);
                 }}
                 disabled={loading}
+                title={account.name}
               >
-                <WalletIcon className="w-6 h-6 -ml-0.5 mr-2 opacity-75 text-gray-700 dark:text-neutral-300" />
-                {account.name}&nbsp;
+                <WalletIcon className="w-6 h-6 -ml-0.5 mr-2 shrink-0 opacity-75 text-gray-700 dark:text-neutral-300" />
+                <div className="overflow-hidden text-ellipsis whitespace-nowrap">
```

Comment:
Could this be a `span`? Block element within this element seems not semantic.

Reply:
Done!